linkerd/app/src/env.rs

```diff
@@ -547,12 +547,18 @@ pub fn parse_config<S: Strings>(strings: &S) -> Result<super::Config, EnvError>
                 Some(addr) => {
                     // If the inbound is proxy is configured to discover policies, then load the set
                     // of all known inbound ports to be discovered during initialization.
-                    let mut ports = match parse(strings, ENV_INBOUND_PORTS, parse_port_set)? {
-                        Some(ports) => ports,
-                        None => {
-                            error!("No inbound ports specified via {}", ENV_INBOUND_PORTS,);
+                    let mut ports = match strings.get(ENV_INBOUND_PORTS)?.as_deref() {
+                        Some("") | None => {
+                            warn!("No inbound ports specified via {}", ENV_INBOUND_PORTS,);
                             Default::default()
                         }
+                        Some(s) => parse_port_set(&s).map_err(|parse_error| {
```

Comment:
`s` is already a reference

```suggestion
                        Some(s) => parse_port_set(s).map_err(|parse_error| {
```

Reply:
@kleimkuhler clippy is still complaining about this ^

Reply:
also I thought we were going to back out this change in favor of a change to parse_port_set